src/en/kickassanime/src/eu/kanade/tachiyomi/animeextension/en/kickassanime/KickAssAnime.kt

```diff
@@ -0,0 +1,299 @@
+package eu.kanade.tachiyomi.animeextension.en.kickassanime
+
+import android.app.Application
+import android.content.SharedPreferences
+import android.util.Base64
+import androidx.preference.ListPreference
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.animesource.ConfigurableAnimeSource
+import eu.kanade.tachiyomi.animesource.model.AnimeFilterList
+import eu.kanade.tachiyomi.animesource.model.AnimesPage
+import eu.kanade.tachiyomi.animesource.model.SAnime
+import eu.kanade.tachiyomi.animesource.model.SEpisode
+import eu.kanade.tachiyomi.animesource.model.Video
+import eu.kanade.tachiyomi.animesource.online.ParsedAnimeHttpSource
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonObject
+import okhttp3.Headers
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import uy.kohesive.injekt.injectLazy
+import java.util.regex.Pattern
+
+@ExperimentalSerializationApi
+class KickAssAnime : ConfigurableAnimeSource, ParsedAnimeHttpSource() {
```

Comment:
```suggestion
class KickAssAnime : ConfigurableAnimeSource, AnimeHttpSource() {
```

your source should be a `AnimeHttpSource`, so you don't have to do this all the time: 
```kotlin
override fun popularAnimeSelector(): String = throw Exception("not used")
```

Reply:
Thanks for the tip 👍
Like the [CONTRIBUTING.MD](https://github.com/jmir1/aniyomi-extensions/blob/master/CONTRIBUTING.md#writing-an-extension) suggested I copied an existing extensionb and started working on it. I choose `nineanime` which happens to be using `ParsedAnimeHttpSource`

Reply:
yep, `ParsedAnimeHttpSource` is mainly used for websites that have most of the necessary information in the HTML tags.

Reply:
Gotcha. 

So, do you have any idea as to why `episodeListParse` is not being reached? my phone won't let me attach a debugger to any process and the logcat is not giving me any useful thing (that I know of 😁)

Should I be worried about this line? It pops up after I open an anime and/or do a refresh
```
2022-08-08 21:38:31.259 18458-22753/? W/hiyomi.mi.debu: Long monitor contention with owner DefaultDispatcher-worker-11 (24098) at okhttp3.Response eu.kanade.tachiyomi.network.interceptor.CloudflareInterceptor.intercept(okhttp3.Interceptor$Chain)(CloudflareInterceptor.kt:17) waiters=0 in okhttp3.Response eu.kanade.tachiyomi.network.interceptor.CloudflareInterceptor.intercept(okhttp3.Interceptor$Chain) for 495ms
```

Reply:
> enable verbose logging at aniyomi's advanced settings

👌👌 I see the problem now, thanks